.github/workflows/release.yml

```diff
@@ -1,58 +1,113 @@
       - name: "Set up Node.js 🧶"
         uses: actions/setup-node@v4
         with:
-          node-version: 18
-          registry-url: "https://registry.npmjs.org"
-          scope: "@conda-store-ui"
+          node-version: "20.x"
+          cache: "yarn"
 
       - name: "Install dependencies 📦"
         run: yarn
 
-      - name: "Lint code 🔎"
+      - name: "Lint code 🔍"
         run: yarn eslint:check
 
-      - name: "Download webpack bundle 📦"
+      - name: "Build application 🏗"
+        run: yarn run build
+
+      - name: "Build application artifacts 🏗"
+        # ensure we use the prod target to minimise assets
+        run: yarn run webpack:prod bundle
+
+      - name: "Generate package tarball 📦"
+        run: yarn pack --filename ${{ env.PACKAGE_FILE }}
+
+      - name: "Upload package tarball 📤"
+        uses: actions/upload-artifact@v4
+        with:
+          name: conda-store-ui-package
+          path: ${{ env.PACKAGE_FILE }}
+
+  release-to-npmjs:
+    name: "Release conda-store-ui to npmjs 📦"
+    runs-on: ubuntu-latest
+    needs: build-application
+    # needed for attestations
+    permissions:
+      id-token: write
+      attestations: write
+      contents: read
+
+    steps:
+      - name: "Checkout repository 🛎"
+        uses: actions/checkout@v4
+
+      # Setup .npmrc file to publish to npm
+      - name: "Set up Node.js 🧶"
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.x"
+          registry-url: "https://registry.npmjs.org"
+          scope: "@conda-store-ui"
+
+      - name: "Download build artefacts 📦"
         uses: actions/download-artifact@v4
         with:
-          name: webpack-bundle
-          path: dist/
+          name: conda-store-ui-package
 
-      - name: "Generate package tarball 📦"
-        run: yarn pack --filename conda-store-ui.tgz
+      - name: "Attest Build Provenance ✨"
+        uses: actions/attest-build-provenance@v1
+        if: github.repository_owner == 'conda-incubator' && github.event_name == 'release' && startsWith(github.ref, 'refs/tags/')
+        with:
+          subject-path: ${{ env.PACKAGE_FILE }}
 
-      - name: "Set NPM scope" #(setup-node workaround https://github.com/actions/setup-node/issues/763)
+      - name: "Set npmjs scope" #(setup-node workaround https://github.com/actions/setup-node/issues/763)
         env:
-          NPM_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
         run: |
           npm config delete @conda-store-ui:registry --location project
           npm config set @conda-store-ui:registry 'https://registry.npmjs.org' --location project
-          npm config set //registry.npmjs.org/:_authToken '${NPM_AUTH_TOKEN}' --location project
+          npm config set //registry.npmjs.org/:_authToken '${NODE_AUTH_TOKEN}' --location project
+
+      - name: "Check npmjs/conda-store-ui scope"
+        run: npm show
+
+      #  we always do a dry run for the publish
```

Comment:
Why?

Reply:
because we want to make sure the actual publish will succeed when we make a release

Reply:
And you either check the logs or the artefacts or both, these are not only useful based on exit codes, but logs that one can go and revisit later to diagnose.
Since we had neither to check we also did not have a quick way to check why/where the `lib` dir has gone missing.

Reply:
I disagree because the same information is going to be printed with `npm publish --verbose`, so this step in the context of a workflow file is 100% redundant. 

I'm sorry to be pushing back so hard, but one reason I think all of this code is so hard to work with is because there's stuff in here and we have no idea why it's here or why it can't be removed. If we're going to keep this step here I think we should document in the comments why it makes sense to do the dry run immediately before the publish, because to me, it makes no sense. In other words, if I come to this workflow file and see this step before the other I will think: this makes no sense, there must be something I don't understand. But I think it just doesn't make sense. It would make sense if there were some code afterwards that grepped the console output for error messages before going to the next step, but we're not doing that here.

Reply:
I am really not privvy of either of these. I added both when refactoring this workflow so if both are confusing/useless we can just remove it. 
As long as we have logs of some sort it is good enough for me.